### ISSUE TITLE
Custom module doesn't update in version 0.9.24

### ISSUE THREAD

**Issue by tamirzb** [Alexays/Waybar#2650]
I have a custom module which prints multiple outputs throughout its run, with the intention of each output updating what is displayed on waybar. It seems that in version 0.9.24 waybar keeps displaying the first output, not updating what is displayed even when other stuff are printed.

To narrow it down, if the script is:
```bash
    echo "{\"text\": \"1\", \"tooltip\": \"tooltip\"," "\"class\": \"class\" }"
    echo "{\"text\": \"2\", \"tooltip\": \"tooltip\"," "\"class\": \"class\" }"
```
I still see `1` printed, not `2`.

On version 0.9.22 this doesn't happend.

**Comment by mxraw** [comment]
I have the same issue. I noticed the bug after updating from Fedora 38 -> 39, which bumped Waybar from 0.9.22 -> 0.9.24.

**Comment by mxraw** [comment]
Bisected this.

```
> git bisect log
git bisect start
# status: waiting for both good and bad commits
# bad: [48ec834ec9c7cdf146169c8b0f669ce1e9a8849e] chore: 0.9.24
git bisect bad 48ec834ec9c7cdf146169c8b0f669ce1e9a8849e
# status: waiting for good commit(s), bad commit known
# good: [e30fba0b8f875c7f35e3173be2b9f6f3ffe3641e] chore: 0.9.22
git bisect good e30fba0b8f875c7f35e3173be2b9f6f3ffe3641e
# bad: [7645ae1907904d7283b1fac70b168e69e52a6f79] Merge pull request #2563 from Syndelis/feat/hyprland-window-rename-by-title
git bisect bad 7645ae1907904d7283b1fac70b168e69e52a6f79
# good: [05a2af2d7c57ce320053b73ed86a58449b5332f1] Merge pull request #2486 from khaneliman/workspace-sorting
git bisect good 05a2af2d7c57ce320053b73ed86a58449b5332f1
# bad: [4c0347d9f243322352a18bdf2c4d4bbcde37163e] Merge pull request #2540 from Syndelis/fix/hyprland-special-workspaces
git bisect bad 4c0347d9f243322352a18bdf2c4d4bbcde37163e
# bad: [f14fe96e199af4bd8050c304ae5ea6c300ba28df] add info to `interval` and `signal` in manpage for custom modules
git bisect bad f14fe96e199af4bd8050c304ae5ea6c300ba28df
# good: [fc6755871763c68005b1c5bd590156596444b967] Merge branch 'master' into darkmode
git bisect good fc6755871763c68005b1c5bd590156596444b967
# good: [4d32991beee8269c72120a10e7c18590017268a6] Merge pull request #2507 from hariseldon78/patch-1
git bisect good 4d32991beee8269c72120a10e7c18590017268a6
# bad: [bf371f70d1e7ea5a38388661d662ee849f46f08e] add `waitingWorker()` to `Custom`
git bisect bad bf371f70d1e7ea5a38388661d662ee849f46f08e
# good: [7c28ffc856a3466269574eae4384269b84c21727] add indefinite `sleep()` function to `SleeperThread`
git bisect good 7c28ffc856a3466269574eae4384269b84c21727
# first bad commit: [bf371f70d1e7ea5a38388661d662ee849f46f08e] add `waitingWorker()` to `Custom`
```

CC @an-prata

**Comment by an-prata** [comment]
No, this is not intended. Right now your modules are being run as though no `restart-interval` has been defined, PR #2517 should not have affected your modules at all, sorry about that. This is fixed in #2778.
